app/scripts/views/mixins/signed-out-notification-mixin.js

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Handle signed-out notifications.
+
+define([
+  'lib/session',
+  'views/base'
+],
+function (Session, BaseView) {
+  'use strict';
+
+  return {
+    initialize: function (options) {
+      this._notifications = options.notifications;
+      this.clearSessionAndNavigateToSignIn = clearSessionAndNavigateToSignIn.bind(this);
+      this._notifications.on(
+        this._notifications.EVENTS.SIGNED_OUT,
+        this.clearSessionAndNavigateToSignIn
+      );
+    },
+
+    destroy: function () {
+      this._notifications.off(
+        this._notifications.EVENTS.SIGNED_OUT,
+        this.clearSessionAndNavigateToSignIn
+      );
+    }
+  };
+
+  function clearSessionAndNavigateToSignIn () {
+    // Unset uid otherwise it will henceforth be impossible
+    // to sign in to different accounts inside this tab.
+    this.relier.unset('uid');
```

Comment:
Future direction for a future PR: now that notifications are sent out and there are several long running objects that can listen for these notifications, the relier could listen for a `SIGNED_OUT` notification and clear the necessary fields. `user` could listen and clear the signed in account UID. formPrefill and session could do the same.